### DISCUSSION TITLE
Unable to deploy Seq to Google Cloud Run

### DISCUSSION THREAD

**Discussion by reuben2crown** [datalust/seq-tickets#1685] [Help] [answered]
Please I have been trying to deploy Seq on google cloud run but every attempt failed with the error below.

`The user-provided container failed to start and listen on the port defined provided by the PORT=5341 environment variable.`

No matter what port I specified, the **Cloud Run** build always return the above error with the specified port number.

docker run --name seq -d --restart unless-stopped -e ACCEPT_EULA=Y -p 5341:8081 datalust/seq:latest
docker tag datalust/seq gcr.io/chisquares/seq
docker push gcr.io/chisquares/seq
gcloud run deploy --image gcr.io/chisquares/seq:latest --max-instances=3 --port 8081

Am I doing anything wrong?

**Comment by liammclennan** [comment, marked as answer]
Hi @reuben2crown ,

I don't have much experience with Cloud Run, but just looking at what you have posted I wonder if the ports may be backwards?

Seq in the container does not run on port `8081` so `-p 5341:8081` won't work. Perhaps you want `-p 8081:5341`?

**Reply by nblumhardt** [reply]
👍 

The container only accepts ingestion on 5341, so just a slight tweak on Liam's suggestion, it'd need to be `8081:80`.

**Reply by reuben2crown** [reply]
I tried both suggestions but still got the same result.

`The user-provided container failed to start and listen on the port defined provided by the PORT=8081 environment variable.`

`The user-provided container failed to start and listen on the port defined provided by the PORT=80 environment variable.`

**Reply by nblumhardt** [reply]
The container normally prints some diagnostic info to STDOUT - do you see any sign of that in your container logs?

**Reply by reuben2crown** [reply]
Thanks for your response. Kindly see the screenshot from the log. I can't seem to figure out what the problem is.

![Screenshot 2022-11-15 123955](https://user-images.githubusercontent.com/7871771/201969185-b4cdc0c0-ab8b-457a-a575-74016ae6eedf.png)
![Screenshot 2022-11-15 123834](https://user-images.githubusercontent.com/7871771/201969216-e48d909f-79d3-4764-991f-67670569b515.png)

**Reply by reuben2crown** [reply]
Kindly see the error log in the attached

**Reply by nblumhardt** [reply]
Thanks! That narrows it down 👍 

Cloud Run is not exposing memory limit information to the container, and Seq's unable to function without it.

It looks like you need to specify the "second generation" Cloud Run execution environment when deploying the service in order to obtain full Linux compatibility: https://cloud.google.com/run/docs/configuring/execution-environments - the docs specifically mention cgroups support (which Seq relies on) as being available in the second generation EE only.

Would be great to hear how you go!

**Reply by nblumhardt** [reply]
(This is in addition to needing to bind to `:80` on the container.)

**Reply by reuben2crown** [reply]
Thank you so much @nblumhardt 

I changed the environment to Second Generation as you advised and it works.

Thank you.
